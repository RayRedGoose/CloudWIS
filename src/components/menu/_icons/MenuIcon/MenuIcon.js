import React from 'react'
import './MenuIcon.scss'

const MenuIcon = ({toggleActiveState}) => {
  const activateMenu = () => {
    toggleActiveState()
  }

  const keyPressed = (event) => {
    if (event.key === "Enter") {
      activateMenu()
    }
  }

  return (
    <svg
      tabIndex = '0'
      role="button"
      onClick={activateMenu}
      onKeyPress={keyPressed}
      x="0px" y="0px"
    	viewBox="0 0 951 951"
      className="menu-icon">
      <path d="M475.5,80.2c-218.3,0-395.3,177-395.3,395.3s177,395.3,395.3,395.3c218.3,0,395.3-177,395.3-395.3 C870.6,257.3,693.7,80.4,475.5,80.2z M475.5,818.1c-189.2,0-342.6-153.4-342.6-342.6s153.4-342.6,342.6-342.6 c189.2,0,342.6,153.4,342.6,342.6C817.9,664.6,664.6,817.9,475.5,818.1z"/>
    	<path d="M422.8,449.1h105.4v52.7H422.8V449.1z"/>
    	<path d="M554.6,449.1h52.7v52.7h-52.7V449.1z"/>
    	<path d="M343.7,449.1h52.7v52.7h-52.7V449.1z"/>
    </svg>
  )
}

export default MenuIcon
