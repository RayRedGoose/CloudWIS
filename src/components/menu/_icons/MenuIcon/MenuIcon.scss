.menu-icon {
  cursor: pointer;
  fill: #017E91;
  width: 40px;
  border-radius: 50%;
  margin: 20px;
  &:hover {
    transform: scale(1.1);
    fill: #ffffff;
  }
  &:active {
    transform: scale(0.9);
  }
}
