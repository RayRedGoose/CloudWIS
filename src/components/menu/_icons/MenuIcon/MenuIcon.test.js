import React from 'react';
import { shallow } from 'enzyme';
import MenuIcon from './MenuIcon';

describe('MenuIcon', () => {
  let icon
  const toggleActiveState = jest.fn()

  beforeEach(() => {
    icon = shallow(
      <MenuIcon toggleActiveState={toggleActiveState} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(icon).toMatchSnapshot();
  });

  it('should call the toggleActiveState prop when clicked', () => {
    icon.simulate('click');

    expect(toggleActiveState).toHaveBeenCalled();
  });

  it('should call the toggleActiveState when enter was pressed', () => {
    icon.simulate('keypress', {key: 'Enter'});

    expect(toggleActiveState).toHaveBeenCalled();
  });

})
