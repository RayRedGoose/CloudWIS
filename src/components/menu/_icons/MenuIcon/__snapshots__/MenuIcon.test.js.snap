// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcon should match the snapshot with all data passed in correctly 1`] = `
<svg
  className="menu-icon"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
  viewBox="0 0 951 951"
  x="0px"
  y="0px"
>
  <path
    d="M475.5,80.2c-218.3,0-395.3,177-395.3,395.3s177,395.3,395.3,395.3c218.3,0,395.3-177,395.3-395.3 C870.6,257.3,693.7,80.4,475.5,80.2z M475.5,818.1c-189.2,0-342.6-153.4-342.6-342.6s153.4-342.6,342.6-342.6 c189.2,0,342.6,153.4,342.6,342.6C817.9,664.6,664.6,817.9,475.5,818.1z"
  />
  <path
    d="M422.8,449.1h105.4v52.7H422.8V449.1z"
  />
  <path
    d="M554.6,449.1h52.7v52.7h-52.7V449.1z"
  />
  <path
    d="M343.7,449.1h52.7v52.7h-52.7V449.1z"
  />
</svg>
`;
